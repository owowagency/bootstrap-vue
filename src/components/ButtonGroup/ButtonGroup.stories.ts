import Btn from '@/components/Button';
import ButtonGroup from '.';
import {sizes} from '@/composables/useSize';

export default {
    title: 'Components/Button group',
    component: ButtonGroup,
    argTypes: {},
};

const template = (args, btnArgs) => ({
    components: {Btn, ButtonGroup},
    setup: () => ({args, btnArgs}),
    template: `
        <ButtonGroup v-bind="args">
            <Btn v-bind="btnArgs">Left</Btn>
            <Btn v-bind="btnArgs">Middle</Btn>
            <Btn v-bind="btnArgs">Right</Btn>
        </ButtonGroup>
    `,
});

export const Default = template.bind({}, {});

export const Outlined = template.bind({}, {}, {outline: true});

export const Sizing = (args) => ({
    components: {Btn, ButtonGroup},
    setup: () => ({args, sizes}),
    template: `
        <ButtonGroup
            v-for="s of sizes"
            class="mb-1 me-1"
            :size="s"
            v-bind="args"
        >
            <Btn>Left</Btn>
            <Btn>Middle</Btn>
            <Btn>Right</Btn>
        </ButtonGroup>
    `,
});

export const Vertical = template.bind({}, {});

Vertical.args = {
    vertical: true,
};
