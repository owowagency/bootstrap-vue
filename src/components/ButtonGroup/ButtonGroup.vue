<template>
    <div :class="classes">
        <!-- @slot Displays content inside the button group -->
        <slot />
    </div>
</template>

<script lang="ts">
import useSize, {sizeProps} from '@/composables/useSize';
import {computed} from 'vue';
import useClasses from '@/composables/useClasses';
</script>

<script lang="ts" setup>
const props = defineProps({
    ...sizeProps,
    /**
     * Stacks the buttons vertically
     */
    vertical: {
        type: Boolean,
        default: false,
    },
});

const {classes} = useClasses(computed(() => [
    computed(() => props.vertical ? 'btn-group-vertical' : 'btn-group').value,
    useSize(props.size, 'btn-group-{0}').sizeClass.value,
]));
</script>

<docs>
```vue
<ButtonGroup vertical>
    <Button>
        Top
    </Button>

    <Button>
        Middle
    </Button>

    <Button>
        Bottom
    </Button>
</ButtonGroup>
```

## Imported Props

| Prop name | Description               | Type                                       | Values | Default |
| --------- | ------------------------- | ------------------------------------------ | ------ | ------- |
| size      | The size of the component | [Size](../../composables/useSize) (string) | -      | 'md'    |
</docs>
