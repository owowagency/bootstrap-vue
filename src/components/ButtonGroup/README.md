# ButtonGroup

```vue
<ButtonGroup vertical>
    <Button>
        Top
    </Button>

    <Button>
        Middle
    </Button>

    <Button>
        Bottom
    </Button>
</ButtonGroup>
```

## Imported Props

| Prop name | Description               | Type                                       | Values | Default |
| --------- | ------------------------- | ------------------------------------------ | ------ | ------- |
| size      | The size of the component | [Size](../../composables/useSize) (string) | -      | 'md'    |

## Props

| Prop name | Description                   | Type    | Values | Default |
| --------- | ----------------------------- | ------- | ------ | ------- |
| vertical  | Stacks the buttons vertically | boolean | -      | false   |

## Slots

| Name    | Description                              | Bindings |
| ------- | ---------------------------------------- | -------- |
| default | Displays content inside the button group |          |
