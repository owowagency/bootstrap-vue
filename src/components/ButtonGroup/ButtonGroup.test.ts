import ButtonGroup from '.';
import {describe} from 'vitest';

describe('template', () => {
    componentRenderTest(ButtonGroup);

    componentSlotRenderTest(ButtonGroup);

    componentWrapperClassTest(ButtonGroup, {vertical: true}, 'btn-group-vertical');

    componentWrapperClassTest(ButtonGroup, {size: 'lg'}, 'btn-group-lg');
});
