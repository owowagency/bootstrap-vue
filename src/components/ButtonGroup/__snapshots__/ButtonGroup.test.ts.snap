// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`template > renders default 1`] = `
"<div class=\\"btn-group btn-group-md\\">
  <!-- @slot Displays content inside the button group -->
</div>"
`;
